packages/schematics/update/update/npm.ts

```diff
@@ -84,7 +84,9 @@ function readOptions(
 
   // Substitute any environment variable references
   for (const key in options) {
-    options[key] = options[key].replace(/\$\{([^\}]+)\}/, (_, name) => process.env[name] || '');
+    if (options[key].replace !== undefined && typeof options[key].replace === 'function') {
```

Comment:
maybe `options[key] && typeof options[key].replace === 'function')` just incase it is undefined/null for some reason?

Reply:
actually we only care about strings here so `typeof options[key] === 'string'` should be sufficient.